lib/widget/button.dart

```diff
@@ -33,6 +37,15 @@ class Button extends StatefulWidget
 
   @override
   Map<String, Function> setters() {
+    final styleModel = GetIt.I<StyleProvider>();
+    final StyleTheme? style = styleModel.getStyleName(_controller.styleName);
+
+    if (style != null) {
+      _controller.color = _controller.color ?? style.color;
```

Comment:
This doesn't scale. None of the widgets should have to know about styleName.

Reply:
This kind of style will be applied to a widget only if it has the `styleName` right? So each widget is supposed to know its styleName right? If it has the styleName, then the style with that name is applied to that widget.

Reply:
@amin-nas i thought we said styleName has no type, and can be applied to any widget.

Reply:
Correct. I believe the ticket here was updated: https://github.com/EnsembleUI/ensemble/issues/432

Reply:
@vusters Okay no need to add type for the widget in styleName but we need to store the styleName for each widget. If it has the styleName, then the style with that name is applied to that widget.